packages/tx/src/eip4844Transaction.ts

```diff
@@ -0,0 +1,479 @@
+    const to = tx.to.value === null ? undefined : Address.fromString(toHexString(tx.to.value))
+    const versionedHashes = tx.blobVersionedHashes.map((el) => Buffer.from(el))
+    const txData = {
+      ...tx,
+      ...{
+        versionedHashes,
+        to,
+        accessList,
+        r: decoded.signature.r,
+        s: decoded.signature.s,
+        v: BigInt(decoded.signature.yParity),
+        gasLimit: decoded.message.gas,
+      },
+    } as BlobEIP4844TxData
+    return new BlobEIP4844Transaction(txData, opts)
+  }
+
+  getUpfrontCost(baseFee: bigint = BigInt(0)): bigint {
+    const prio = this.maxPriorityFeePerGas
+    const maxBase = this.maxFeePerGas - baseFee
+    const inclusionFeePerGas = prio < maxBase ? prio : maxBase
+    const gasPrice = inclusionFeePerGas + baseFee
+    return this.gasLimit * gasPrice + this.value
+  }
+
+  raw(): TxValuesArray {
+    throw new Error('Method not implemented.')
+  }
+
+  /**
+   * Serialize a blob transaction to the execution payload variant
+   * @returns the minimum (execution payload) serialization of a signed transaction
+   */
+  serialize(): Buffer {
+    const to = {
+      selector: this.to !== undefined ? 1 : 0,
+      value: this.to?.toBuffer() ?? null,
+    }
+    const sszEncodedTx = SignedBlobTransactionType.serialize({
+      message: {
+        chainId: this.common.chainId(),
+        nonce: this.nonce,
+        maxPriorityFeePerGas: this.maxPriorityFeePerGas,
+        maxFeePerGas: this.maxFeePerGas,
+        gas: this.gasLimit,
+        to,
+        value: this.value,
+        data: this.data,
+        accessList: this.accessList.map((listItem) => {
+          return { address: listItem[0], storageKeys: listItem[1] }
+        }),
+        blobVersionedHashes: this.versionedHashes,
+        maxFeePerDataGas: this.maxFeePerDataGas,
+      },
+      // TODO: Decide how to serialize an unsigned transaction
+      signature: {
+        r: this.r ?? BigInt(0),
+        s: this.s ?? BigInt(0),
+        yParity: this.v === BigInt(1) ? true : false,
+      },
+    })
+    return Buffer.concat([TRANSACTION_TYPE_BUFFER, sszEncodedTx])
+  }
+
+  serializeNetworkWrapper(): Buffer {
+    const to = {
+      selector: this.to !== undefined ? 1 : 0,
+      value: this.to?.toBuffer() ?? null,
+    }
+    const blobArrays = this.blobs?.map((blob) => Uint8Array.from(blob)) ?? []
+    const serializedTxWrapper = BlobNetworkTransactionWrapper.serialize({
+      blobs: blobArrays,
+      blobKzgs: this.kzgCommitments?.map((commitment) => Uint8Array.from(commitment)) ?? [],
+      tx: { ...this.txData(), ...to },
+      kzgAggregatedProof: kzg.computeAggregateKzgProof(blobArrays),
+    })
+    return Buffer.concat([Buffer.from([0x05]), serializedTxWrapper])
+  }
+
+  getMessageToSign(hashMessage: false): Buffer | Buffer[]
+  getMessageToSign(hashMessage?: true | undefined): Buffer
+  getMessageToSign(_hashMessage?: unknown): Buffer | Buffer[] {
+    return this.hash()
+  }
+
+  hash(): Buffer {
```

Comment:
Just to note: inherited and overwritten methods need some code docs as well at some point, these are not taken from the upper class (e.g. in VSCode).